Feature - Header  -#9

larstp opened the pull request:
## Description

This pull request adds a header element that can be used on all pages.
Note that the Figma design prototype didn't link properly to everything so I have added things where it was missing.
I've tested it on 4 different screens including mobile and iPad, and it should work on everything but please give it a try.

Note: the mobile dropdown menu coming in from the right and covering the entire screen is deliberate, as it is how it is done in the Figma file (this confused me in the beginning). :)

## To test:

### Steps:

Open /public/index.html with live server.

Open DevTools (press F12 or right-click → Inspect).

Go to the Console tab.

Paste and run one of the following commands:

Log in as student: `localStorage.setItem('user', JSON.stringify({ username: 'student', role: 'student' }));`

Log in as teacher: `localStorage.setItem('user', JSON.stringify({ username: 'teacher', role: 'teacher' }));`

Log in as admin: `localStorage.setItem('user', JSON.stringify({ username: 'admin', role: 'admin' }));`

Refresh the page to see the header/menu update for that user type.

--

To log out: `localStorage.removeItem('user');`

Then refresh the page.



---

## Related Issue

Closes #9 

---

## Changes Made

- [X] Added Header for desktop and mobile
- [X] Added README in /components on how to insert
- [X] Matched all styling to figma design, including spacing, icon + text sizes etc. 
- [X] Finished header.js + header.html
- [X] Added styling for Header in components.css
- [X] Added media query for changing between desktop and mobile header in the media.css file.
- [X] Deleted deprecated CSS folder
- [X] Added JSON file in the /api folder to use as mockup

---

## Screenshots (if applicable)

![image](https://github.com/user-attachments/assets/168aa524-69fd-44b2-b84e-c312b9bd3d44)
---

## Acceptance Criteria


- [x] Changes have been tested locally (mobile + desktop)
- [x] Follows the style guide and naming conventions (e.g. kebab-case)
- [x] Matches design spec (e.g. from Figma)
- [x] No `console.log` or warnings
- [x] All images have `alt` text
- <s> [ ] Headings follow correct order (no skipping levels)

emmelinlarina commented on src/css/components.css:
```diff
@@ -0,0 +1,189 @@
+  font-family: 'Poppins', sans-serif;
+  font-weight: 600;
+  font-size: 20px;
+  line-height: 7%;
```
The 7% line-height might cause weird spacing. Maybe change it to 1.2 - 1.4?

emmelinlarina commented on src/css/components.css:
```diff
@@ -0,0 +1,189 @@
+  pointer-events: none;
+}
+
+.close-dropdown {
```
.hamburger and .close-dropdown could be together, since they both have the same style (to DRY it up)

emmelinlarina commented on src/css/media.css:
```diff
@@ -0,0 +1,49 @@
+  }
+
+  .nav-links a {
+    list-style: none;
```
This can go in .nav-links <li> since its not doing anything on the <a>

emmelinlarina commented on src/pages/about.html:
This file is missing the font you imported to `<index.html>`. Maybe move the font import to the global css? Or add it here for now

larstp replied:
Added.

emmelinlarina commented on src/css/components.css:
```diff
@@ -0,0 +1,189 @@
+  font-family: 'Poppins', sans-serif;
+  font-weight: 600;
+  font-size: 20px;
+  line-height: 7%;
```
The 7% line-height might cause weird spacing. Maybe change it to 1.2 - 1.4?

larstp replied:
You're right! I followed the Figma board but 7% does cause some problems. Changed to 1.2

emmelinlarina commented on src/css/components.css:
```diff
@@ -0,0 +1,189 @@
+  pointer-events: none;
+}
+
+.close-dropdown {
```
.hamburger and .close-dropdown could be together, since they both have the same style (to DRY it up)

larstp replied:
I'm blind. You are right. Merged!

emmelinlarina commented on src/css/media.css:
```diff
@@ -0,0 +1,49 @@
+  }
+
+  .nav-links a {
+    list-style: none;
```
list-style can go in .nav-links `<li>`, since its not doing anything on the `<a>`

larstp replied:
Done!

emmelinlarina commented on src/pages/contact.html:
This file is also missing the font you imported to <index.html>. Maybe move the font import to the global css? Or add it here for now

larstp replied:
Added.

emmelinlarina commented on src/script/api/README.md:
Add this under Usage of mockup JSON file in readme: 

This mock login setup is only for local development and testing.
Do **not** use this code for real authentication or production use.

larstp replied:
Added.

emmelinlarina commented on src/script/utils/header.js:
```diff
@@ -0,0 +1,320 @@
+  logoLink.href = '/public/index.html';
+
+  const logo = document.createElement('img');
+  logo.src = '/public/assets/img/logo-header.png';
```
Absolute paths like `</public/assets/...>` might cause issues in production, depending on how the site is hosted. It works fine for now, but something to keep in mind later.

larstp replied:
Yes! I've made a note to change that at a later date, as we're still in the process of creating the site and some things might move around. 

I'll change the paths closer to handoff when we have finished more pages :)

emmelinlarina commented on src/script/utils/header.js:
This is looking very good! Although there might be some repetition which makes the code longer, but at long as the code works I don't think its a problem for now. Unless you want to go over it some time later. :D

larstp replied:
Yeah, I thought about creating more functions outside of the main two to separate and speed up the process, as well as keeping it more DRY, but I feel it might have to wait until we're all more comfortable with JS. :p 
It turned very advanced very quickly when I tried it..

emmelinlarina replied:
Very understandable!

emmelinlarina commented on src/css/components.css:
Very good! Looking clean and DRY! I just have to minor things for you to look at:D

emmelinlarina commented on src/css/media.css:
Superb!

emmelinlarina reviewed (commented):
### Superb! 
Wow! It looks very clean!! Almost everything is superDRY:D I've marked some very minor things.

emmelinlarina commented:
I think everything is ready from my side! Just need to click "ready for review" so we can merge it in 👍

larstp commented:
Fixed all issues and added footer. 
Had to test on a page so made the About page to test on that one, so now we have that i guess. :p 

Let me know if there is more i need to fix :)
